turkey/job.py

```diff
@@ -89,8 +89,10 @@ def run(self, stdout=False):
             # the case where tasks are pinned to fewer than cpu_count number of
             # cores
             if self.intelligent:
-                args['threads'] = int(
-                    mp.cpu_count() / config.num_tasks_in_system)
+                task.args['nthreads'] = int(
```

Comment:
So, there was a bug, but this fix isn't right. What we want is to replace args['threads'] with args['nthreads']. task.args['nthreads'] may not be populated (although I'm guessing in your case it is) and we also want to distribute the number of cores among the tasks in system, not the number of threads you wanted to run 'task' with among tasks in the system.